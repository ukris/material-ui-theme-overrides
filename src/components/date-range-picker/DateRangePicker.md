DateRangePicker Standard Example:

```jsx
    <DateRangePicker
        format='MMM dd'
        dates={{start: "2020-08-08T03:00:00.000Z", end:"2020-08-23T03:00:00.000Z"}}
        updateCell={
            (datesArray) => alert(`Sent to Parent Component:\n start: ${datesArray[0]}\n end: ${datesArray[1]}`)
        }
    />
```