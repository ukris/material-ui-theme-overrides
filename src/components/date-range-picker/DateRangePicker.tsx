import React from 'react'
import { DesktopDateRangePicker, LocalizationProvider } from "@material-ui/pickers"
import enLocale from "date-fns/locale/en-US"
import DateFnsAdapter from "@material-ui/pickers/adapter/date-fns"
const localeMap = {
    en: enLocale
}

interface DateRangePickerProps {
    /** Stringified dates */
    dates: {start?: string, end?: string},
  
    /** Date format*/
    format: string,

    /** Updates row data in parent component with the purpose of persisting when switching table page */
    updateCell:  (datesArray: Date[]) => void
}

const onChange = (date:any) => {
    // updateCell
}

export default ({dates, format, updateCell} : DateRangePickerProps) => {
    const {start, end} = dates;
    return (
        <LocalizationProvider dateAdapter={DateFnsAdapter} locale={localeMap['en']}>
            <DesktopDateRangePicker
                data-id="daterangepicker"
                calendars={1}
                open={true}
                minDate={new Date("1900-01-01")}
                renderInput={(_startProps, _endProps) => (<></>)}
                value={[
                    start ? new Date(start) : new Date(), 
                    end ? new Date(end) : new Date()
                ]}
                onChange={onChange}
            />
        </LocalizationProvider>
    )
}
