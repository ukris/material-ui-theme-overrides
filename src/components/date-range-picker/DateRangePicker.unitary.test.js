import React from 'react'
import renderer from 'react-test-renderer'
import { mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import DateRangePicker from './DateRangePicker'

configure({ adapter: new Adapter() })


describe('DateRangePicker', () => {
    const component = (
        <DateRangePicker
            format='MMM dd'
            dates={{start: "2020-08-08T03:00:00.000Z", end:"2020-08-23T03:00:00.000Z"}}
            updateRow={
                (datesArray) => alert(`Sent to Parent Component:\n start: ${datesArray[0]}\n end: ${datesArray[1]}`)
            }
        />
    )
    const wrapper = mount(component)
    const datepickerComponent = wrapper.find('[data-test="table-daterangepicker"]')

    it('renders component correctly', () => {
        const formattedDate = wrapper.find('input')
        expect(datepickerComponent.length).not.toEqual(0)
        expect(formattedDate.length).toEqual(1)
        expect(formattedDate.props().value).toEqual('Aug 08 - Aug 23')
    });

    it('matches snapshot', () => {
        const tree = renderer.create(component,).toJSON()
        const formattedDate = tree.children[0].children[0].children[0].props.value

        expect(tree).toMatchSnapshot()
        expect(formattedDate).toEqual('Aug 08 - Aug 23')
    });
})
