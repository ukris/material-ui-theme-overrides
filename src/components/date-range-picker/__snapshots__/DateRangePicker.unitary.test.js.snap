// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangePicker matches snapshot 1`] = `
<div
  className="table-daterangepicker"
>
  <div
    className="MuiFormControl-root MuiTextField-root"
    data-test="table-daterangepicker"
    onClick={[Function]}
    onKeyPress={[Function]}
    open={true}
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        mask={null}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        pipe={null}
        readOnly={true}
        required={false}
        type="text"
        value="Aug 08 - Aug 23"
      />
    </div>
    
  </div>
</div>
`;
